Types/Identifier.php

```diff
@@ -83,6 +83,14 @@ public static function column(array $tokens, MetaData $metaData) {
 
         $idColumns  = !empty($meta) ? end($meta)->getIdentifierColumnNames() : [];
 
-        return !empty($idColumns) ? end($idColumns) : 'id';
+        if (empty($idColumns)) {
```

Comment:
First of all: Thank you for contributing! And sorry for being that late.

In general your bugfix is a good one. But we just need to clean up the code a bit. If you had a look at the rest of the code we tried...
... not to mutate existing variables and
... not to create/mutate variables in a branch

Example:
```php
    if($x) {
        $var = 'value';
    } else {
        $var = 'anotherValue';
    }
```
turns to
```php
$var = $x ? 'value' : 'anotherValue';
```

The idea is to use conditions as statements instead of expressions to ensure we don't generate side effects.

So what we need here is something like:
```php
    if(!empty($idColumns)) return end($idColumns);

    $reservedTokens = array_filter($tokens['SELECT'], function($token) {
        return $token['expr'] === 'reserved';
    });

    $finalTokens = count($reservedTokens) > 0 ? $reservedTokens : $tokens;

    return end($finalTokens[0]['no_quotes']['parts']);
```
Maybe I missed a detail, but I hope it helps you to understand what I want to change here.
May you be so kind and change these few lines so I can merge the PR?